src/StackExchange.Redis/LogProxy.cs

```diff
@@ -1,50 +1,77 @@
 ﻿using System;
 using System.IO;
+using Microsoft.Extensions.Logging;
 
 namespace StackExchange.Redis;
 
 internal sealed class LogProxy : IDisposable
 {
-    public static LogProxy? TryCreate(TextWriter? writer)
-        => writer == null ? null : new LogProxy(writer);
+    public static LogProxy? TryCreate(TextWriter? writer, ConfigurationOptions? options)
+        => writer == null && options?.Logger == null
+        ? null
+        : new LogProxy(writer, options?.Logger);
 
-    public override string ToString()
+    private TextWriter? _log;
+    private ILogger? _logger;
+    internal static Action<string> NullWriter = _ => { };
+    public object SyncLock => this;
+
+    private LogProxy(TextWriter? log, ILogger? logger)
     {
-        string? s = null;
-        if (_log != null)
+        _log = log;
+        _logger = logger;
+    }
+
+    public void WriteLine()
+    {
+        if (_log is not null) // note: double-checked
         {
             lock (SyncLock)
             {
-                s = _log?.ToString();
+                _log?.WriteLine();
             }
         }
-        return s ?? base.ToString() ?? string.Empty;
     }
-    private TextWriter? _log;
-    internal static Action<string> NullWriter = _ => { };
 
-    public object SyncLock => this;
-    private LogProxy(TextWriter log) => _log = log;
-    public void WriteLine()
+    public void LogInfo(string? message = null)
     {
+        var msg = $"{DateTime.UtcNow:HH:mm:ss.ffff}: {message}";
         if (_log != null) // note: double-checked
```

Comment:
Maybe we could consider exacting a method to write with `_log`, maybe

``` csharp
private void WriteForTextWriter(string? message)
{
  if (_log is null) return;
  lock (SyncLock)
  {
    if (message is null)
    {
      _log?.WriteLine();
    }
    else
    {
      _log?.WriteLine(message);
    }
  }
}
```

Reply:
Thanks for the suggestion! In this case I'm not sure it cleans it up much, because we're calling a different overload each time (trace/info/error/etc.) and the only 2 that stare a signature are Trace & Info each of which would take a action for the proper trace/info method underneath. If we could use it for more it might be worth breaking out, but not sure we'd save much here...better to be consistent across all methods I think :)